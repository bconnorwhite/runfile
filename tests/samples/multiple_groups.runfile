# ----------
# Development
# ----------

# Build the project
build
  --debug     # Enable debug mode
  --release   # Build in release mode
  --target=<target>: # Target platform
  echo "Building for $target"

# Run tests
test
  --verbose   # Verbose output
  --coverage  # Generate coverage report
  --filter=<pattern>: # Test filter pattern
  echo "Running tests with filter: $filter"

# ----------
# Deployment
# ----------

# Deploy to staging
deploy-staging
  --force     # Force deployment
  --dry-run:  # Dry run mode
  echo "Deploying to staging"

# Deploy to production
deploy-prod
  --force     # Force deployment
  --backup:   # Create backup
  echo "Deploying to production"

# ----------
# Utilities
# ----------

# Clean up
clean
  --all       # Clean all
  --cache:    # Clean cache
  echo "Cleaning up"

# Show help
help:
  echo "Available commands:"
