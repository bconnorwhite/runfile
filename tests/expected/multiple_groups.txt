Development
  build           # Build the project
    --debug         # Enable debug mode
    --release       # Build in release mode
    --target        # Target platform
  test            # Run tests
    --verbose       # Verbose output
    --coverage      # Generate coverage report
    --filter        # Test filter pattern

Deployment
  deploy-staging  # Deploy to staging
    --force         # Force deployment
    --dry-run       # Dry run mode
  deploy-prod     # Deploy to production
    --force         # Force deployment
    --backup        # Create backup

Utilities
  clean           # Clean up
    --all           # Clean all
    --cache         # Clean cache
  help            # Show help

